src/renderer/html_handlebars/hbs_renderer.rs

```diff
@@ -650,28 +652,92 @@ pub fn normalize_id(content: &str) -> String {
 }
 
 /// Uses elasticlunr to create a search index and exports that into `searchindex.json`.
-fn make_searchindex(book: &MDBook, search_documents : Vec<utils::SearchDocument>) -> Result<()> {
-    let mut index = elasticlunr::index::Index::new("id",
-        &["title".into(), "body".into(), "breadcrumbs".into()]);
+fn make_searchindex(book: &MDBook,
+                    search_documents : Vec<utils::SearchDocument>,
+                    searchconfig : &Search) -> Result<()> {
 
-    for sd in search_documents {
-        let anchor = if let Some(s) = sd.anchor.1 {
-            format!("{}#{}", sd.anchor.0, &s)
-        } else {
-            sd.anchor.0
-        };
 
-        let mut map = HashMap::new();
-        map.insert("id".into(), anchor.clone());
-        map.insert("title".into(), sd.title);
-        map.insert("body".into(), sd.body);
-        map.insert("breadcrumbs".into(), sd.hierarchy.join(" » "));
-        index.add_doc(&anchor, map);
+    #[derive(Serialize)]
+    struct SearchOptionsField {
+        boost: u8,
+    }
+
+    #[derive(Serialize)]
+    struct SearchOptionsFields {
+        title: SearchOptionsField,
+        body: SearchOptionsField,
+        breadcrumbs: SearchOptionsField,
+    }
+
+    /// The searchoptions for elasticlunr.js
+    #[derive(Serialize)]
+    struct SearchOptions {
+        bool: String,
```

Comment:
Just out of curiosity, what does this guy do?

Reply:
This mimics the structure of the `searchconfig` object, that elasticlunr expects. At first i serialized the `config::Search` struct, but that uses kebab-case which is not nice and would cause hidden breakage if modified.

Reply:
Ah fair enough, so it's a elasticlunr-specific thing then. I thought `bool` was an unusual choice for a variable in a Rust program, given it's also the name of a primitive type.